lib/utils/report.mjs

```diff
@@ -279,7 +279,7 @@ function computeEditInfo({ fix, line, result: { stylelint }, ruleName }) {
 	// Compute the smallest range and text of the fix
 	const fixData = narrowFixRange(node, {
 		range: fixedNodeRange,
-		text: node.toString(),
+		text: node.parent?.raws.semicolon ? `${node.toString()};` : node.toString(),
```

Comment:
I think that this needs tests.
i.e. I think that it probably assumes something too specific regarding the position of the child node inside the parent
e.g. what if it's not last?